### PR TITLE
Resolving conflict in issue #582

### PR DESCRIPTION
**Pull request by Gasu16**
---
#### TODO

- [x] Use `cargo fmt`
- [x] Add necessary tests
- [x] Add changelog entry

### CODE REVIEW

**Comment by zwpaper** [comment]
Hi @Gasu16, thanks for the contribution, but it seems the `-d` is not working with `-a` perfectly, there is something that has to do to align with the `GNU ls`.

your implementation:

```console
$ cargo run -- -a -d
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/lsd -a -d`
 .   ..   .git   .github   .gitignore   .release.toml   build.rs   Cargo.lock   Cargo.toml   CHANGELOG.md   ci   CODEOWNERS   doc   LICENSE   README.md   src   target   tests
```

GNU ls:

```console
$ \ls -a -d
.
```

without `-a`:

```console
$ \ls -d
.

$ lsd -d
 .
```

show directories:

```console
$ \ls -d -a */
ci/     doc/    src/    target/ tests/

$ cargo run -- -d -a */
    Finished dev [unoptimized + debuginfo] target(s) in 0.27s
     Running `target/debug/lsd -d -a ci/ doc/ src/ target/ tests/`

ci/:
 .   ..   before_deploy.bash   before_install.bash   script.bash

doc/:
 .   ..   lsd.md

src/:
 .   ..   app.rs   color   color.rs   config_file.rs   core.rs   display.rs   flags   flags.rs   icon.rs   main.rs   meta   sort.rs

target/:
 .   ..   .rustc_info.json   CACHEDIR.TAG   debug

tests/:
 .   ..   integration.rs
```

**Comment by zwpaper** [comment]
Hi @Gasu16, thanks for the contribution, but it seems the `-d` is not working with `-a` perfectly, there is something that has to do to align with the `GNU ls`.

your implementation:

```console
$ cargo run -- -a -d
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/lsd -a -d`
 .   ..   .git   .github   .gitignore   .release.toml   build.rs   Cargo.lock   Cargo.toml   CHANGELOG.md   ci   CODEOWNERS   doc   LICENSE   README.md   src   target   tests
```

GNU ls:

```console
$ \ls -a -d
.
```

without `-a`:

```console
$ \ls -d
.

$ lsd -d
 .
```

show directories:

```console
$ \ls -d -a */
ci/     doc/    src/    target/ tests/

$ cargo run -- -d -a */
    Finished dev [unoptimized + debuginfo] target(s) in 0.27s
     Running `target/debug/lsd -d -a ci/ doc/ src/ target/ tests/`

ci/:
 .   ..   before_deploy.bash   before_install.bash   script.bash

doc/:
 .   ..   lsd.md

src/:
 .   ..   app.rs   color   color.rs   config_file.rs   core.rs   display.rs   flags   flags.rs   icon.rs   main.rs   meta   sort.rs

target/:
 .   ..   .rustc_info.json   CACHEDIR.TAG   debug

tests/:
 .   ..   integration.rs
```

**Comment by Gasu16** [comment]
@zwpaper thank you for pointing that, gonna send a new PR soon

**Comment by Gasu16** [comment]
Ok so I corrected, I did some tests and it seems the output is aligned with `GNU ls`, here I show some examples:
### My implementation
```
$ cargo run -- -a -d
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/lsd -a -d`
 .

$ lsd -a -d
 .
```

### GNU ls output
```
$ ls -a -d
.
```

### Show directories
```
$ cargo run -- -a -d */
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/lsd -a -d ci/ doc/ src/ target/ tests/`
 ci/   doc/   src/   target/   tests/
```

```
$ ls -a -d */
ci/  doc/  src/  target/  tests/
```

### With -ahl option
```
$ cargo run -- -ahl --color -d */
    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
     Running `target/debug/lsd -ahl --color -d ci/ doc/ src/ target/ tests/`
drwxr-xr-x matt matt 4.0 KB Mon Nov 22 19:07:14 2021  ci/
drwxr-xr-x matt matt 4.0 KB Mon Nov 22 19:07:14 2021  doc/
drwxr-xr-x matt matt 4.0 KB Mon Nov 22 19:07:14 2021  src/
drwxr-xr-x matt matt 4.0 KB Tue Nov 23 14:24:56 2021  target/
drwxr-xr-x matt matt 4.0 KB Tue Nov 23 15:57:53 2021  tests/
```
```
$ ls -ahl --color -d */
drwxr-xr-x 2 matt matt 4.0K Nov 22 19:07 ci/
drwxr-xr-x 2 matt matt 4.0K Nov 22 19:07 doc/
drwxr-xr-x 5 matt matt 4.0K Nov 22 19:07 src/
drwxr-xr-x 4 matt matt 4.0K Nov 23 14:24 target/
drwxr-xr-x 2 matt matt 4.0K Nov 23 15:57 tests/
```

**Comment by Gasu16** [comment]
@zwpaper Hi, I saw 5 checks out of 14 failed, what should I do to fix them?

File: src/app.rs [hunk 1]
```diff
@@ -120,8 +120,8 @@ pub fn build() -> App<'static, 'static> {
             Arg::with_name("directory-only")
                 .short("d")
                 .long("directory-only")
-                .conflicts_with("all")
-                .conflicts_with("almost-all")
+                //.conflicts_with("all")
```
Comment (meain) [inline src/app.rs:123]: There is no reason to keep the code commented. We can complete drop it.

File: src/flags/display.rs [hunk 2]
```diff
@@ -26,7 +26,11 @@ impl Configurable<Self> for Display {
     /// [None].
     fn from_arg_matches(matches: &ArgMatches) -> Option<Self> {
         if matches.is_present("all") {
-            Some(Self::All)
+            if matches.is_present("directory-only") {
```
Comment (meain) [inline src/flags/display.rs:29]: This can be better accomplished by just flipping the order of the if conditions.

**Review by meain — CHANGES_REQUESTED** [review]

File: src/app.rs (hunk 1 above)
Comment (Gasu16) [reply src/app.rs:123]: You right, gonna remove it.

File: src/flags/display.rs (hunk 2 above)
Comment (Gasu16) [reply src/flags/display.rs:29]: You mean to check first "directory-only" and then the if "all" nested?

**Review by Gasu16 — COMMENTED** [review]
Edited according to the review suggested

**Comment by zwpaper** [comment]
@Gasu16 Thanks for contributing!

**Pull request merged**